src/components/datasets/records/GraphBrowser/GraphBrowse2.vue

```diff
@@ -0,0 +1,355 @@
+<script setup>
+import { nextTick, ref } from 'vue'
+import { VueFlow, useVueFlow } from '@vue-flow/core'
+import { Background } from '@vue-flow/background'
+import { ControlButton, Controls } from '@vue-flow/controls'
+import { MiniMap } from '@vue-flow/minimap'
+
+
+
+import ModelsList from "@/components/datasets/records/ModelsList/ModelsList.vue";
+import IconArrowRight from "@/components/icons/IconArrowRight.vue";
+import * as site from '../../../../site-config/site.json';
+import { MarkerType } from '@vue-flow/core'
+import { useLayout } from './useLayout'
+
+import {useHandleXhrError, useSendXhr} from '@/mixins/request/request_composable'
+
+import { computed } from 'vue';
+import { useStore } from 'vuex';
+import {pathOr} from "ramda";
+import {useRoute} from "vue-router";
+const route = useRoute()
+
+import CustomNode from './ModelNode.vue'
+import IconUnlink from "@/components/icons/IconUnlink.vue";
+import IconSizeToFit from "@/components/icons/IconSizeToFit.vue";
+
+const store = useStore()
+
+const hasModels = computed( () => store.state.concepts.length > 0)
+/**
+ * `useVueFlow` provides:
+ * 1. a set of methods to interact with the VueFlow instance (like `fitView`, `setViewport`, `addEdges`, etc)
+ * 2. a set of event-hooks to listen to VueFlow events (like `onInit`, `onNodeDragStop`, `onConnect`, etc)
+ * 3. the internal state of the VueFlow instance (like `nodes`, `edges`, `viewport`, etc)
+ */
+const { onInit, onNodeDragStop, onConnect, addEdges, setViewport, toObject, fitView } = useVueFlow()
+
+let edges = ref()
+let nodes = ref()
```

Comment:
We might want to make sure assigning these as reactive refs and then reassigning them later is the right way to go about this - I think that can cause unnecessary renders. I'll look into other approaches for when I build the workflow manager and see if there is another way to do this - but this clearly works well for now.

Reply:
Okay, yeah, still a little new to the composition api. I think we should supply a default value maybe